tools/python/mwm_downloader.py

```diff
@@ -0,0 +1,135 @@
+        return False
+    return True
+
+
+def progress_bar(total, progress):
+    """
+    Displays or updates a console progress bar.
+
+    Original source: https://stackoverflow.com/a/15860757/1391441
+    """
+    initial_progress = progress
+    bar_length = 20
+    progress = float(progress) / float(total)
+    block = int(round(bar_length * progress))
+    text = '\r[{}] {:.0f}% {}/{}'.format(
+        '#' * block + '-' * (bar_length - block), round(progress * 100, 0), initial_progress, total)
+    sys.stdout.write(text)
+    if progress == 1.:
+        sys.stdout.write('\n')
+    sys.stdout.flush()
+
+
+def download_mwm_list(revision, mwm_list, threads, folder, silent=True):
+    possible_revisions = get_possible_mwm_revisions()
+
+    if revision is None:
+        # Get latest revision by default
+        revision = max(possible_revisions)
+    elif revision not in possible_revisions:
+        logger.error('{revision} is not found in {url}'.format(revision=revision, url=DIRECT_MAP_URL))
+        return 1
+
+    possible_mwm_names = get_possible_mwm_names(revision)
+    if mwm_list is not None:
+        for mwm_name in mwm_list:
+            if mwm_name not in possible_mwm_names:
+                logger.error('{mwm_name} map is not found in {url}'.format(mwm_name=mwm_name, url=DIRECT_MAP_URL))
+                return 1
+    else:
+        mwm_list = possible_mwm_names
+
+    folder = folder if folder is not None else '{base}/{revision}'.format(base=DEFAULT_DOWNLOAD_DIRECTORY,
+                                                                          revision=revision)
+
+    if not os.path.exists(folder):
+        os.makedirs(folder)
+
+    # No sense to show progress bar for one map
+    if len(mwm_list) < 2:
+        silent = True
+
+    not_downloaded_maps = []
+    with ThreadPoolExecutor(max_workers=threads) as executor:
+        future_to_mwm = {}
+        for mwm_name in mwm_list:
+            unescaped_name = mwm_name.replace(' ', '%20')
+            url = '{base}{revision}/{mwm_name}.mwm'.format(base=DIRECT_MAP_URL, revision=revision, mwm_name=unescaped_name)
+            filename = '{folder}/{mwm_name}.mwm'.format(folder=folder, mwm_name=mwm_name)
+            future_to_mwm[executor.submit(download_file, url, filename)] = mwm_name
+
+        for mwm_number, future in enumerate(as_completed(future_to_mwm)):
+            if not future.result():
+                not_downloaded_maps.append(future_to_mwm[future])
+            if not silent:
+                progress_bar(len(future_to_mwm), mwm_number + 1)
+
+    if not_downloaded_maps:
+        logger.error('Not loaded maps: ')
+        [logger.error(' - {name}'.format(name=name)) for name in not_downloaded_maps]
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Script to download \'.mwm\' files in multiply threads')
+    parser.add_argument('-r', '--revision', help='Map revision number e.g. 180126', type=int, default=None)
```

Comment:
It is map version technically. Up to you.

Reply:
Fixed.